### PR TITLE
Iter20

### PR DESCRIPTION
**Pull request by npavlov**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -11,12 +11,17 @@ all: fmt lint build-agent build-server
 # Build the server binary from Go source files in cmd/server directory
 .PHONY: build-server
 build-server:
-	$(GO) build -gcflags="all=-N -l" -o bin/server ${CURDIR}/cmd/server/*.go
+	$(GO) build -gcflags="all=-N -l" -ldflags="-X 'main.buildVersion=1.0.0' -X 'main.buildDate=$(shell date -u +%Y-%m-%d)' -X 'main.buildCommit=$(shell git rev-parse --short HEAD)'" -o bin/server ${CURDIR}/cmd/server/*.go
```
Comment (arttet) [inline Makefile:14]:
```suggestion
	$(GO) build -gcflags="all=-N -l" -ldflags="-X 'main.buildVersion=1.0.0' -X 'main.buildDate=$(shell date -u +%Y-%m-%d)' -X 'main.buildCommit=$(shell git rev-parse  HEAD)'" -o bin/server ${CURDIR}/cmd/server/*.go
```

File: cmd/agent/main.go [hunk 2]
```diff
@@ -18,20 +18,45 @@ import (
 	"github.com/npavlov/go-metrics-service/internal/utils"
 )
 
+var (
+	buildVersion = "N/A"
```
Comment (arttet) [inline cmd/agent/main.go:22]: You can move it to the suitable package. For example, `version`.

File: cmd/agent/main.go [hunk 3]
```diff
@@ -1,9 +1,9 @@
+//nolint:gochecknoglobals
```
Comment (arttet) [inline cmd/agent/main.go:1]: Please fix it. This shouldn't be used for the whole package.

File: cmd/agent/main.go [hunk 4]
```diff
@@ -18,20 +18,45 @@ import (
...
+
+	cfg := loadConfig(&log)
+
+	ctx, cancel := utils.WithSignalCancel(context.Background(), &log)
+	defer cancel()
+
+	runAgent(ctx, cfg, &log)
+}
+
+func setupLogger() zerolog.Logger {
```
Comment (arttet) [inline cmd/agent/main.go:40]: Please simplify the main package.

File: pkg/analysers/exitchecker.go [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+	Name: "exitcheck",
+	Doc:  "check for calling os.Exit()",
+	Run:  run,
+}
+
+//nolint:gocognit,cyclop,nonamedreturns
+func run(pass *analysis.Pass) (res any, err error) {
+	for _, file := range pass.Files {
+		haveSpecImport := false
+		ast.Inspect(file, func(node ast.Node) bool {
```
Comment (arttet) [inline pkg/analysers/exitchecker.go:28]: You can simplify this function.

File: pkg/analysers/testdata/src/notmainfunc/code.go [hunk 6]
```diff
@@ -0,0 +1,8 @@
+package main
+
+import "os"
+
+// No error should be reported here because there is no Exit in main function
```
Comment (arttet) [inline pkg/analysers/testdata/src/notmainfunc/code.go:5]: I would like to get a warning in this corner case.

File: pkg/analysers/testdata/src/notmainpackage/code.go [hunk 7]
```diff
@@ -0,0 +1,8 @@
+package test
+
+import "os"
+
+// No error should be reported here because there is no Exit in main package
```
Comment (arttet) [inline pkg/analysers/testdata/src/notmainpackage/code.go:5]: I would like to get a warning in this corner case.

File: pkg/analysers/testdata/src/notmainfunc/code.go (hunk 6 above)
Comment (arttet) [inline pkg/analysers/testdata/src/notmainfunc/code.go:5]: Please add more tests.

**Review by arttet — APPROVED** [review]

File: cmd/agent/main.go (hunk 2 above)
Comment (npavlov) [reply cmd/agent/main.go:22]: fixed

File: pkg/analysers/exitchecker.go (hunk 5 above)
Comment (npavlov) [reply pkg/analysers/exitchecker.go:28]: simplified

File: Makefile (hunk 1 above)
Comment (npavlov) [reply Makefile:14]: added full commit
